File: exercises/ex01/code.py
# code.py

import threading

i = 0

def task_a():
    global i
    
    for j in range(0, 1000000):
        i += 1

def task_b():
    global i

    for j in range(0, 1000000):
        i -= 1

thread1 = threading.Thread(target=task_a)
thread2 = threading.Thread(target=task_b)

thread1.start()
thread2.start()

thread1.join()
thread2.join()

print i

File: exercises/ex01/code.go
package main

import "fmt"

var i = 0

func task_a(ch chan int) {
	for j := 0; j < 1000000; j++ {
		i++
	}
	ch <- 0
}

func task_b(ch chan int) {
	for j := 0; j < 1000000; j++ {
		i--
	}
	ch <- 0
}

func main() {

	//task_a()
	//task_b()

	ch := make(chan int)
	go task_a(ch)
	go task_b(ch)

	<-ch
	<-ch

	fmt.Println(i)

}
